### ISSUE TITLE
Combat Tracker: Adding entries with counting values

### ISSUE THREAD

**Issue by mckee14** [Plushtoast/dsa5-foundryVTT#135]
It would massively help being able to:

- add individual entries to the combat tracker (basically just a name with a INI and a value)
- assign an INI manually for that entry, and/or assign the entry to a char (linking it, so when the chars INI changes, so does the entrys INI)
- Assign a numerical value to the entry
- Allow the numerical value to automatically increase / decrease after turns/rounds, set by the user or source

Example:
"Donald" is casting an Armatrutz, it will be active for 5 rounds. So the GM can add the entry "Armatrutz" (maybe even linked to Donald), set 5 rounds duration. Every time a round is over, it adds "-1" to the value. When it reaches 0, the Armatrutz is over.

Possible features / adaptions:
- Linking the entry to one or several chars, allowing:
  - Automatic creation of an entry when using a spell/item, or being targeted by a spell
  - Entry can store modifications (like +1GS, +2TP), influencing the linked char(s)
- The entry can be configured to run the calculation for the value at the start of a round, end of a round, or when the entry itself is activated(or skipped) in the combat tracker
- de-activating an entry when it reaches 0, maybe with a confirmation from the GM

Variants:
- implementation via status instead of combat tracker entries.
  - Temporary effects can be posted to chat and shown in the combat tracker (see module "[active effects to chat](https://foundryvtt.com/packages/ae-to-chat/)" for possible inspiration)
  - would allow to check for certain status (f.e: a char is preparing a spell for 3 KR, setting the status "active spellcasting". when he gets attacked and hit, the module could ask for a roll to keep up the concentration, according to the GRW).

All in all, it would help the GM being able to "set and forget" certain things in a combat, without forgetting them completely. You don't need a reminder that the char has 3RS, because the module accounts for that automatically. But you need to remember that he has +2RS extra, but only for 3 more rounds.

**Comment by Plushtoast** [comment]
So the plan for this is:
Make status effects affect everything (not only roll modifiers but also attributes)
Then make them easily accessible (drag drop)
Then allow any custom effect, which can be timegated at least per combat rounds but maybe also with real time or a time module.

So only after the last step we will think about modifying the combat tracker. but that might not be needed anymore at this point because it is calculated automatically.

**Comment by Nyhles** [comment]
There is already a Module that does the first part (entries in Combat Tracker with Ini and decreasing duration): [Combat Effects Tracker](https://github.com/MKamysz/combat-effects-tracker).

Although it currently is not perfectly compatible, at least for me I cant open the Settings Tab when the Module is activated.
Still, it certainly can be a big inspiration.

But as discussed in Issue #101 

> Make status effects affect everything (not only roll modifiers but also attributes)
Then make them easily accessible (drag drop)
Then allow any custom effect, which can be timegated at least per combat rounds but maybe also with real time or a time module.

it can be handled with dynamic active effects and would be a real gamechanger.

**Comment by mckee14** [comment]
I think this is one of these "highly important, but low priority" things. Finding the right way to approach this topic depends on many factors currently only known by the devs and ulisses - what is planned to be released as official products, what is the planned feature-set, do you accept dependency on other modules you can't control yourself?

@Plushtoast Your idea sounds great so far and would allow to add temporary effects in combat situations via stat... status? stati? statuses? damn you, english! question is, how this would possibly integrate with other things in the future. Take Berittener Kampf as an example: Is "Sitting on a horse" a status for the char? What about "it is dark around you, sight modifier -3". Could this be an environmental modifier, applied to all chars affected by it? Would a lantern elimininate 2 of these -3 modifiers when equipped?

Personally, I would let that sink in for some days and possibly check with ulisses, how complete and complex they want this whole thing to become.

**Comment by Plushtoast** [comment]
This is mostly concluded with the last steps being #320

**Comment by DerGreif** [comment]
You might want to look at the Turn Alert module for Foundry: https://foundryvtt.com/packages/turnAlert/

It allows multiple turn trackers (as count downs) for each combatant and multiple general round trackers. You get a chat message, when they have counted down and you can even add a macro, which will be executed at the end of the turn count down. I use the macro primarily to give a sound alert, because otherwise the chat message is easily missed in the heat of combat. 

Anyway it might help OP to aid until some native TDE functionality is established or might give some ideas to the devs.

You are doing great work! It is highly appreciated! I cannot stress this enough!

Kind regards,
G.

**Comment by Plushtoast** [comment]
The last thing I am going to add here is progress of spell channelling.
I think this is enough to cover at least more than 80% of use cases (Active Effects is magic for everything).
